src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -797,16 +822,62 @@ public void run() {
                 return Executor.of(this);
             }
 
-            @Restricted(NoExternalUse.class) // for Jelly
+            @Restricted(NoExternalUse.class) // for Jelly and toString
             public String getUrl() {
                 return PlaceholderTask.this.getUrl(); // we hope this has a console.jelly
             }
 
+            @Exported(name="url")
+            public String getAbsoluteUrl() {
+                Run<?,?> r = runForDisplay();
+                if (r == null) {
+                    return "";
+                }
+                Jenkins j = Jenkins.getInstance();
+                String base = "";
+                if (j != null) {
+                    base = Util.removeTrailingSlash(j.getRootUrl()) + "/";
+                }
+                return base + r.getUrl();
+            }
+
             @Override public String toString() {
                 return "PlaceholderExecutable:" + PlaceholderTask.this;
             }
 
             private static final long serialVersionUID = 1L;
+
+            @Override
+            public ACL getACL() {
+                try {
```

Comment:
I think it would be better to just use `return getParent().getACL();` since the goal is to use identical logic.